### ISSUE TITLE
Support loading resource modules from custom folder

### ISSUE THREAD

**Issue by andersmelander** [andersmelander/better-translation-manager#39]
It should be possible to have resource modules loaded from a custom folder instead of only supporting load from the application folder.

Context: https://en.delphipraxis.net/topic/1865-ann-better-translation-manager-released/?do=findComment&comment=72784

**Comment by BastianTeufel-GTR** [comment]
Hi and thanks for sharing this awesome project.
Would it be possible for you to implement it?

I made the modifications accordingly in my local version but when updating your repo the changes would be gone.

I've implemented it like written by you. It's a change in a single function.
I modified it to look for two different subfolders: `lang` and `l10n'

Any chance that this feature will be implemented as it shouldn't be a big change.
I can also make a PR if you want.

**Comment by andersmelander** [comment]
@BastianTeufel-GTR No need for a PR; I've already looked at your change.

I'm not too thrilled with hard-coding the subfolder names so I'll be looking into how to do it in a way that will allow the paths to be externally configured.

**Comment by andersmelander** [comment]
@BastianTeufel-GTR Please try the solution in the [ResourceModuleSubFolders](https://github.com/andersmelander/better-translation-manager/tree/ResourceModuleSubFolders) branch.

You can specify the resource module paths using the ResourceModulePaths variable:
https://github.com/andersmelander/better-translation-manager/blob/e89401bc0eab17612cd87e49b1794208208dd6df/Source/amLanguageInfo.pas#L232-L237

Like this:
```pascal
ResourceModulePaths := 'lang;L10n';
```

Note that I haven't actually verified that it works. I'll leave that up to you :-)

**Comment by BastianTeufel-GTR** [comment]
@andersmelander thanks for the implementation. 
You're right, hard coding is not the best option here.

I think using a variable to set the path is a good option. In that way one who want's to implement specific paths could do so in a wrapper and set the path accordingly.

I'll checkout the mentioned branch, test it and let you know.

Best regards
Bastian

**Comment by BastianTeufel-GTR** [comment]
I did some first tests with implementing the call to `LocalizationTools.LoadResourceModule('ENU')`
using the variable as implemented by you works as expected perfectly fine.

The only issue I found is the following scenario, based on the HelloWorld example project:
- Native Application is English (so compiled in the exe)
- Language files have been moved to a subdirectory like `l10n`
- The user has no special language selected like `da-DK`, so I want to let the program decide to load the language based on the Windows System language e.g. `de-DE`

But in that case the language files are not found, because the internal mechanism is not looking into the specified subfolders.

So when I want to move the files to a subdirectory I **ALWAYS** need to call the LoadResourceModule function.
In advance, I manually need to fetch  the Windows locale to be able to pass the Windows Locale or a custom language to the LoadResourceModule function. Otherwise the ResourceModulePath variable is not evaluated.

**Comment by andersmelander** [comment]
> So when I want to move the files to a subdirectory I ALWAYS need to call the LoadResourceModule function.

I'm afraid there's no way around that.
The functions that the RTL automatically call to locate and load resource modules (`DelayLoadResourceModule`, `GetModuleFileName`, `LoadResourceModule`, etc.) does not provide us with any hooks or any way to specify the location of the resource modules; They only allow us to specify the desired language via a registry key.

I _could_ solve that by run-time patching the RTL code with detours but I think that is beyond the scope of the project and not worth the risk and added maintenance.

**Comment by BastianTeufel-GTR** [comment]
I agree, fiddling around with the Delphi internals will be out of scope of this project and could lead to addition problems.
